### DISCUSSION TITLE
Install Chain Code failed in Kubernetes v1.27

### DISCUSSION THREAD

**Discussion by ylinam** [hyperledger/fabric#4837] [Q&A]
![image](https://github.com/hyperledger/fabric/assets/115536031/2fcc75ae-2083-461e-ac34-48c0f212bbf6)

I tried to install chain code in Kubernetes v1.27, but got the following error
```shell
chaincode install failed with status: 500 - failed to invoke backing implementation of 'InstallChaincode': 
could not build chaincode: docker build failed: docker image inspection failed: 
Get "http://unix.sock/images/dev-service-org1.0.2-5ac7d45868cb07b81a9319dc0cedd017595e267b0815c32076bd4410f90b090d-6d3088fec5f40817a7bf70bf7a2cc03091982623daa66f46e6108c5d20d63f03/json": 
net/http: HTTP/1.x transport connection broken: malformed HTTP response "\x00\x00\x06\x04\x00\x00\x00\x00\x00\x00\x05\x00\x00@\x00"
``` 

The hyperledger-fabric version is 2.2.1. 
The value of CORE_VM_ENDPOINT is set to unix:///run/containerd/containerd.sock

Moreoever, I tried the hyperledger-orderer version 2.3 and set the OREDER ADMIN parameters, but got the same result.

```
          - name: ORDERER_ADMIN_LISTENADDRESS
            value: 0.0.0.0:7051
          - name: ORDERER_ADMIN_TLS_ENABLED
            value: "true"
          - name: ORDERER_ADMIN_TLS_PRIVATEKEY
            value: /var/hyperledger/orderer/tls/server.key
          - name: ORDERER_ADMIN_TLS_CERTIFICATE
            value: /var/hyperledger/orderer/tls/server.crt
          - name: ORDERER_ADMIN_TLS_ROOTCAS
            value: "[/var/hyperledger/orderer/tls/ca.crt]"
          - name: ORDERER_METRICS_PROVIDER
            value: prometheus
```

**Comment by satota2** [comment]
@ylinam
For running Fabric in a Kubernetes environment, you should use the Chaincode as a Service feature.
Using this feature should help you avoid the issues you are experiencing with the chaincode installation failure.
You can find the details here:
https://hyperledger-fabric.readthedocs.io/en/latest/cc_service.html

For a Kubernetes-based example using KIND, check out this sample which employs the Chaincode as a Service model.
https://github.com/hyperledger/fabric-samples/tree/main/test-network-k8s

Additionally, Fabric v2.2 and 2.3 are beyond their community maintenance period. I strongly recommend using  v2.5 (the latest LTS version) or later, to ensure better support and stability.
